pkg/controllers/multiclusterservice/mcs_controller.go

```diff
@@ -463,3 +612,12 @@ func (c *MCSController) SetupWithManager(mgr controllerruntime.Manager) error {
 		WithOptions(controller.Options{RateLimiter: ratelimiterflag.DefaultControllerRateLimiter(c.RateLimiterOptions)}).
 		Complete(c)
 }
+
+func needSyncMultiClusterService(mcs *networkingv1alpha1.MultiClusterService, clusterName string) bool {
+	if len(mcs.Spec.ServiceProvisionClusters) == 0 || len(mcs.Spec.ServiceConsumptionClusters) == 0 {
+		return true
+	}
+	clusters := sets.New[string](mcs.Spec.ServiceProvisionClusters...)
```

Comment:
There is util function `helper.GetProvisionClusters` and `helper.GetConsumptionClustres`.